src/backends/plonky2/circuits/mainpod.rs

```diff
@@ -133,6 +144,54 @@ impl OperationVerifyGadget {
         Ok(())
     }
 
+    fn eval_contains_from_entries(
+        &self,
+        builder: &mut CircuitBuilder<F, D>,
+        st: &StatementTarget,
+        op: &OperationTarget,
+        resolved_merkle_claim: MerkleClaimTarget,
+        resolved_op_args: &[StatementTarget],
+    ) -> BoolTarget {
+        let op_code_ok = op.has_native_type(builder, NativeOperation::ContainsFromEntries);
+
+        let arg_types_ok = self.first_n_args_are_valueofs(builder, 3, resolved_op_args);
+
+        let merkle_root_value = resolved_op_args[0].args[1].as_value();
+        let key_value = resolved_op_args[1].args[1].as_value();
+        let value_value = resolved_op_args[2].args[1].as_value();
```

Comment:
What do you think about merging these two operations into one?  Imagine this method
```rust
impl OperationVerifyGadget {
    fn first_n_args_as_valueofs<const N: usize>(
        &self,
        builder: &mut CircuitBuilder<F, D>,
        resolved_op_args: &[StatementTarget],
    ) -> (BoolTarget, [ValueTarget; N]) {
        // ...
    }
}
```
then you can call it like this:
```rust
        let (arg_types_ok, [merkle_root_value, key_value, value_value]) =
            self.first_n_args_as_valueofs::<3>(builder, resolved_op_args);
```